botcore/regex.py

```diff
@@ -0,0 +1,21 @@
+"""Common regular expressions."""
+
+import re
+
+DISCORD_INVITE = re.compile(
+    r"(discord([.,]|dot)gg|"                     # Could be discord.gg/
+    r"discord([.,]|dot)com(/|slash)invite|"      # or discord.com/invite/
+    r"discordapp([.,]|dot)com(/|slash)invite|"   # or discordapp.com/invite/
+    r"discord([.,]|dot)me|"                      # or discord.me
+    r"discord([.,]|dot)li|"                      # or discord.li
+    r"discord([.,]|dot)io|"                      # or discord.io.
+    r"((?<!\w)([.,]|dot))gg"                     # or .gg/
+    r")([/]|slash)"                              # / or 'slash'
+    r"(?P<invite>[a-zA-Z0-9\-]+)",               # the invite code itself
+    flags=re.IGNORECASE
+)
+"""
+Regex for discord server invites.
+
+:meta hide-value:
```

Comment:
Is this for PyCharm?

Reply:
No it's for Sphinx. If we didn't add this, the docs would have the entire regex written to it. You can see a list of all flags in the Sphinx documentation.